tests/test_helper.hpp

```diff
@@ -74,8 +89,13 @@ bool almost_equal(T1<T> x, T2<T> y, int ulp) {
                  std::is_same<T2<T>, sycl::ext::cplx::complex<T>>::value),
                 "almost_equal should be used to compare complex number");
   // Somebody should be smart enough to refactor to use `enable_if`...
-  return almost_equal(x.real(), y.real(), ulp) &&
-         almost_equal(x.imag(), y.imag(), ulp);
+  auto diff =
+      complex_magnitude(std::complex(x.real() - y.real(), x.imag() - y.imag()));
```

Comment:
Please replace with  `std::abs(a-b)`. Hence I don't think we need the `complex_magnitude` anymore

Reply:
The reason I did this was because x and y can be different complex types, like from std vs sycl, and so it fails if you try and directly perform some arithmetic. I have fixed this by casting one of the types. So it is now `std::abs((T2<T>)x - y)`

Reply:
Oh, I see, do you know why your implicit cast didn't work in this case? I was thinking that did all the machinery to get those thing working

Reply:
I am not 100% sure, the issue appears to be that no implicit conversion is taking place. So it is a bit odd. However, even if the implicit conversion could be made to work this way, it would probably fail due to it being ambiguous as to which operator it should use. It would have to select between `operator -(std::complex, std::complex)` and `operator -(sycl::complex, sycl::complex)` both would require 1 conversion and there would not be a clear one to chose.

I think if we want to have `std::complex` and `sycl::complex` working with each other this way then it would be best to explicitly state the operations to avoid issues with template and function ambiguity.

Compilation error:
```
/usr/lib/gcc/x86_64-linux-gnu/9/../../../../include/c++/9/complex:361:5: note: candidate template ignored: could not match 'std::complex' against 'sycl::ext::cplx::complex'
    operator-(const complex<_Tp>& __x, const complex<_Tp>& __y)
    ^
/usr/lib/gcc/x86_64-linux-gnu/9/../../../../include/c++/9/complex:370:5: note: candidate template ignored: could not match 'std::complex' against 'sycl::ext::cplx::complex'
    operator-(const complex<_Tp>& __x, const _Tp& __y)
    ^
/home/aidan/SyclCPLX/include/sycl_ext_complex.hpp:775:1: note: candidate template ignored: could not match 'sycl::ext::cplx::complex' against 'std::complex'
operator-(const complex<_Tp> &__x, const complex<_Tp> &__y) {
^
/home/aidan/SyclCPLX/include/sycl_ext_complex.hpp:791:1: note: candidate template ignored: could not match 'sycl::ext::cplx::complex' against 'std::complex'
operator-(const _Tp &__x, const complex<_Tp> &__y) {
^
```